bitcoin/src/blockdata/transaction.rs

```diff
@@ -1173,6 +1171,30 @@ impl From<&Transaction> for Wtxid {
     fn from(tx: &Transaction) -> Wtxid { tx.wtxid() }
 }
 
+/// Returns the effective_value `Amount` returning `None` on overflow.
+///
+/// The effective value is the value of this output minus the amount to spend it.  That is, the
+/// effective_value can be calculated as value - (fee_rate * weight).
+///
+/// Note: the effective value of a [`Transaction`] may increase more than the effective value of
+/// a [`TxOut`] when adding another [`TxOut`] to the transaction.  This happens when the new
+/// [`TxOut`] added causes the output length `VarInt` to increase its encoding length.
+///
+/// # Arguments
+///
+/// * `fee_rate` - `FeeRate`in sat/kwu
+/// * `satisfaction_weight` - satisfied spending conditions weight
+// Note, this method could return Result instead of Option for richer errors.
```

Comment:
This comment is useless. And no, I don't think `Result` is useful here.

Reply:
The only possible error is overflow. I don't think you can improve much on it. But if you have a real experience of debugging overflows I'd look into it.

Reply:
Yes but there's three possible reasons for overflow that might be valuable information to bubble back?

Reply:
removed comment, marking resolved.